daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -156,12 +156,15 @@ final class ImmArray[+A] private (
   }
 
   /** O(n) */
-  def reverse: ImmArray[A] = {
-    val newArray: mutable.ArraySeq[A] = new mutable.ArraySeq(len)
-    for (i <- indices) {
-      newArray(i) = array(start + len - (i + 1))
+  lazy val reverse: ImmArray[A] = {
```

Comment:
reverse is very rarely called. We probably do not want to use extra memory to cache this result. 
Note that reverse is scala immutable collection is also defined as `def`

Reply:
agreed